libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -339,65 +341,17 @@ internal class MapboxTripSession(
-                    "state: ${status.routeState}",
-                LOG_CATEGORY
-            )
-            logD(LOG_CATEGORY) {
-                "navigatorObserver#onStatus; banner instruction: [${status.bannerInstruction}]," +
-                    " voice instruction: [${status.voiceInstruction}]"
-            }
-
-            val tripStatus = status.getTripStatusFrom(primaryRoute)
-            val enhancedLocation = tripStatus.navigationStatus.location.toLocation()
-            val keyPoints = tripStatus.navigationStatus.keyPoints.toLocations()
-            val road = RoadFactory.buildRoadObject(tripStatus.navigationStatus)
-            updateLocationMatcherResult(
-                tripStatus.getLocationMatcherResult(enhancedLocation, keyPoints, road)
-            )
-            zLevel = status.layer
-
-            // we should skip RouteProgress, BannerInstructions, isOffRoute state updates while
-            // setting a new route
-            if (isUpdatingRoute) {
-                logD("route progress update dropped - updating routes", LOG_CATEGORY)
-                return
-            }
-
-            var triggerObserver = false
-            if (tripStatus.navigationStatus.routeState != RouteState.INVALID) {
-                val nativeBannerInstruction = tripStatus.navigationStatus.bannerInstruction
-                val bannerInstructions =
-                    tripStatus.navigationStatus.getCurrentBannerInstructions(primaryRoute)
-                triggerObserver = bannerInstructionEvent.isOccurring(
-                    bannerInstructions,
-                    nativeBannerInstruction?.index
-                )
-            }
-            val remainingWaypoints = tripStatus.calculateRemainingWaypoints()
-            val latestBannerInstructionsWrapper = bannerInstructionEvent.latestInstructionWrapper
-            val routeProgress = getRouteProgressFrom(
-                tripStatus.route,
-                tripStatus.navigationStatus,
-                remainingWaypoints,
-                latestBannerInstructionsWrapper?.latestBannerInstructions,
-                latestBannerInstructionsWrapper?.latestInstructionIndex,
-                lastVoiceInstruction
-            ).also {
-                if (it == null) {
-                    logD(
-                        "route progress update dropped - " +
-                            "currentPrimaryRoute ID: ${primaryRoute?.id}; " +
-                            "currentState: ${status.routeState}",
-                        LOG_CATEGORY
-                    )
+            try {
+                processNativeStatus(status)
+            } catch (error: Throwable) {
+                logE(LOG_CATEGORY) {
+                    "Error processing native status update: origin=$origin, status=$status.\n" +
+                        "Error: $error\n" +
+                        "MapboxTripSession state: " +
+                        "isUpdatingRoute=$isUpdatingRoute, primaryRoute=${primaryRoute?.id}"
```

Comment:
I'm wondering, does it make sense to move this text to exception message and don't log it manually 🤔

Reply:
I think makes sense to leave it as a log message in case if the exception will be swallowed in the upstream code.